pool/src/events.rs

```diff
@@ -0,0 +1,326 @@
+    /// * from - The address whose position is being modified
+    /// * tokens_in - The amount of tokens sent to the pool
+    /// * b_tokens_minted - The amount of b_tokens minted
+    pub fn supply_collateral(
+        e: &Env,
+        asset: Address,
+        from: Address,
+        tokens_in: i128,
+        b_tokens_minted: i128,
+    ) {
+        let topics = (Symbol::new(e, "supply_collateral"), asset, from);
+        e.events().publish(topics, (tokens_in, b_tokens_minted));
+    }
+
+    /// Emitted when collateral is withdrawn
+    ///
+    /// - topics - `["withdraw_collateral", asset: Address, from: Address]`
+    /// - data - `[tokens_out: i128, b_tokens_burnt: i128]`
+    ///
+    /// ### Arguments
+    /// * asset - The asset
+    /// * from - The address whose position is being modified
+    /// * tokens_out - The amount of tokens withdrawn from the pool
+    /// * b_tokens_burnt - The amount of b_tokens burnt
+    pub fn withdraw_collateral(
+        e: &Env,
+        asset: Address,
+        from: Address,
+        tokens_out: i128,
+        b_tokens_burnt: i128,
+    ) {
+        let topics = (Symbol::new(e, "withdraw_collateral"), asset, from);
+        e.events().publish(topics, (tokens_out, b_tokens_burnt));
+    }
+
+    /// Emitted when tokens are borrowed
+    ///
+    /// - topics - `["borrow", asset: Address, from: Address]`
+    /// - data - `[tokens_out: i128, d_tokens_minted: i128]`
+    ///
+    /// ### Arguments
+    /// * asset - The asset
+    /// * from - The address whose position is being modified
+    /// * tokens_out - The amount of tokens sent from the pool
+    /// * d_tokens_burnt - The amount of d_tokens burnt
+    pub fn borrow(e: &Env, asset: Address, from: Address, tokens_out: i128, d_tokens_minted: i128) {
+        let topics = (Symbol::new(e, "borrow"), asset, from);
+        e.events().publish(topics, (tokens_out, d_tokens_minted));
+    }
+
+    /// Emitted when a loan is repaid
+    ///
+    /// - topics - `["repay", asset: Address, from: Address]`
+    /// - data - `[tokens_in: i128, d_tokens_burnt: i128]`
+    ///
+    /// ### Arguments
+    /// * asset - The asset
+    /// * from - The address whose position is being modified
+    /// * tokens_in - The amount of tokens sent to the pool
+    /// * d_tokens_burnt - The amount of d_tokens burnt
+    pub fn repay(e: &Env, asset: Address, from: Address, tokens_in: i128, d_tokens_burnt: i128) {
+        let topics = (Symbol::new(e, "repay"), asset, from);
+        e.events().publish(topics, (tokens_in, d_tokens_burnt));
+    }
+
+    /// Emitted when a new auction is created
+    ///
+    /// - topics - `["new_auction", user: Address, auction_type: u32]`
+    /// - data - `auction_data: AuctionData`
+    ///
+    /// ### Arguments
+    /// * user - The auction user
+    /// * auction_type - The type of auction
+    /// * auction_data - The auction data
+    pub fn new_auction(e: &Env, user: Address, auction_type: u32, auction_data: AuctionData) {
```

Comment:
do we want to emit the percent liquidated as well?

Reply:
Liquidation percent is not auction agnostic, so it is currently left out.

While the data is potentially useful, it is also indexable via transactions, given liquidation percent is an argument, and auction data is the return value.